imagehash.go

```diff
@@ -60,11 +61,52 @@ func (h *ImageHash) GetHash() uint64 {
 }
 
 // GetKind method returns a kind of image hash.
-func (h *ImageHash) GetKind() hashKind {
+func (h *ImageHash) GetKind() Kind {
 	return h.kind
 }
 
 // Set method sets a bit of index.
 func (h *ImageHash) Set(idx int) {
 	h.hash |= 1 << uint(idx)
 }
+
+const strFmt = "%v:%0x"
+
+// FromString returns an image hash from a hex representation
+func FromString(s string) (*ImageHash, error) {
+	var kindStr string
+	var hash uint64
+	_, err := fmt.Sscanf(s, strFmt, kindStr, hash)
+	if err != nil {
+		return nil, errors.New("Couldn't parse string " + s)
+	}
+
+	kind := Unknown
+	switch kindStr {
+	case "a":
+		kind = AHash
+	case "p":
+		kind = PHash
+	case "d":
+		kind = DHash
+	case "w":
+		kind = WHash
+	}
+	return NewImageHash(hash, kind), nil
+}
+
+// ToString returns a hex representation of the hash
+func (h *ImageHash) ToString() string {
```

Comment:
Can you add unit tests to this function also?
You can create NewImageHash from `FromString`and then check the string value is identical.

Reply:
Added some tests there! Also, renamed `FromString` to `ImageHashFromString`.